be/src/vec/functions/function_datetime_string_to_string.h

```diff
@@ -117,4 +178,85 @@ class FunctionDateTimeStringToString : public IFunction {
     }
 };
 
+template <typename Transform>
+class FunctionDateTimeStringToStringFourArgs : public IFunction {
+public:
+    static constexpr auto name = Transform::name;
+    static constexpr bool has_variadic_argument =
+            !std::is_void_v<decltype(has_variadic_argument_types(std::declval<Transform>()))>;
+
+    static FunctionPtr create() { return std::make_shared<FunctionDateTimeStringToStringFourArgs>(); }
+
+    String get_name() const override { return name; }
+
+    size_t get_number_of_arguments() const override { return 4; }
+    bool is_variadic() const override { return true; }
+    DataTypes get_variadic_argument_types_impl() const override {
+        if constexpr (has_variadic_argument) return Transform::get_variadic_argument_types();
+        return {};
+    }
+
+    DataTypePtr get_return_type_impl(const ColumnsWithTypeAndName& arguments) const override {
+        return make_nullable(std::make_shared<DataTypeString>());
+    }
+
+    bool use_default_implementation_for_nulls() const override { return false; }
+    bool use_default_implementation_for_constants() const override { return true; }
+    ColumnNumbers get_arguments_that_are_always_constant() const override { return {1, 2, 3}; }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) override {
+        const ColumnPtr source_col = block.get_by_position(arguments[0]).column;
+
+        const auto* nullable_column = check_and_get_column<ColumnNullable>(source_col.get());
+        const auto* sources = check_and_get_column<ColumnVector<typename Transform::FromType>>(
+                nullable_column ? nullable_column->get_nested_column_ptr().get()
+                                : source_col.get());
+
+        if (sources) {
+            auto col_res = ColumnString::create();
+            ColumnUInt8::MutablePtr col_null_map_to;
+            col_null_map_to = ColumnUInt8::create();
+            auto& vec_null_map_to = col_null_map_to->get_data();
+
+            if(arguments.size() == 4) {
+                GET_COLUMN_CONST_RETURN_IF_ERROR(1, ColumnConst);
+                GET_COLUMN_CONST_RETURN_IF_ERROR(2, ColumnConst);
+                GET_COLUMN_CONST_RETURN_IF_ERROR(3, ColumnConst);
+
+                Int32 first_day_of_week = delta_const_column1->get_field().get<Int32>();
```

Comment:
`delta_const_column##x` changed. then change this to elimate virtual function call.